packages/apm/src/integrations/tracing.ts

```diff
@@ -336,11 +343,171 @@ export class Tracing implements Integration {
     const active = Tracing._activeTransaction as SpanClass;
     if (active) {
       logger.log('[Tracing] finishIdleTransaction', active.transaction);
+      Tracing._addPerformanceEntries(active);
       // true = use timestamp of last span
       active.finish(true);
     }
   }
 
+  /**
+   * This uses `performance.getEntries()` to add additional spans to the active transaction.
+   * Also, we update our timings since we consider the timings in this API to be more correct than our manual
+   * measurements.
+   *
+   * @param transactionSpan The transaction span
+   */
+  private static _addPerformanceEntries(transactionSpan: SpanClass): void {
+    let navigationOffset = 0;
+    if (
+      transactionSpan.op === 'navigation' &&
+      transactionSpan.data &&
+      typeof transactionSpan.data.offset === 'number'
+    ) {
+      navigationOffset = transactionSpan.data.offset;
+    }
+    // tslint:disable-next-line: completed-docs
+    function addSpan(span: SpanClass): void {
+      // tslint:disable-next-line: no-non-null-assertion
+      transactionSpan.spanRecorder!.finishSpan(span);
+    }
+
+    // tslint:disable-next-line: completed-docs
+    function addPerformanceNavigationTiming(parent: SpanClass, entry: any, event: string): void {
+      const span = parent.child({
+        description: event,
+        op: 'browser',
+      });
+      // tslint:disable: no-unsafe-any
+      span.startTimestamp = parent.startTimestamp + Tracing._msToSec(entry[`${event}Start`] as number);
```

Comment:
It can be written as `span.timestamp = span.startTimestamp = parent.startTimestamp + Tracing._msToSec(entry[`${event}Start`] as number);`

Reply:
Nope, 
`Tracing._msToSec(entry[`${event}Start`]`
`Tracing._msToSec(entry[`${event}End`]`